spynnaker/pyNN/models/common/abstract_neuron_recordable.py

```diff
@@ -107,3 +107,20 @@ def get_expected_n_rows(
         :param str variable: the variable being recorded
         :return: int the number of rows expected.
         """
+
+    @abstractmethod
+    def get_rewires(
```

Comment:
I think this got missed in the updates.  I wondered here if instead of get_rewires, you could call get_data("rewires".....  This would then avoid downstream changes, and is more consistent with the rest of the variables.

Reply:
I am not sure about this - spikes don't go through get_data("spikes"), for example, as far as I can tell?  The reason I did it this way was to avoid going through get_matrix_data(...), which is not what we want to do for rewires/events.  But I suppose we could go into the vertex's get_data(..) function and then go through to the neuron_recorder's get_rewires(..) function, which removes the need for a get_rewires() function in the vertex - is that what you mean?

Reply:
At this level spikes do go through get_data.  It is only the level below that factors this out into the spikes etc.

Reply:
There's a get_spikes(...) function in the vertex - that's the level I'm talking about.  This comes from AbstractSpikeRecordable, rather than AbstractNeuronRecordable, though.  The only level where get_data("spikes") happens is at the Population level, I think - once you then get into the recorder's append_current_segment(...), it splits into get_spikes() and get_recorded_matrix(...).

Reply:
I don't think AbstractNeuronRecordable has get_spikes.  It might be that this is a mistake if things are already calling that!

Reply:
This is true, get_spikes is in AbstractSpikeRecordable.  Perhaps we need AbstractEventRecordable as well?  Then this can be added to vertices as and when necessary (thus avoiding the downstream changes in e.g. SpiNNGym).